lib/matplotlib/widgets.py

```diff
@@ -1417,6 +1417,543 @@ def _update(self):
 
 
+class Ruler(AxesWidget):
+    """
+    A ruler to measure distances and angles on an axes instance. 
+
+    For the ruler to remain responsive you must keep a reference to it.
+
+     Parameters
+    ----------
+
+    *ax*  : the  :class:`matplotlib.axes.Axes` instance 
+
+    *active* : bool, default is True
+        Whether the ruler is active or not. 
+
+    *length_unit*  : string, a unit identifier to use in displayed text  
+        i.e. ('ft', or 'm')
+
+    *angle_unit*  : "degrees" or "radians"
+        The type of angle unit ('degrees' or 'radians')
+
+    *print_text*  : bool, default is False
+        Whether the length measure string is printed to the console
+
+    *useblit* : bool, default is False
+        If True, use the backend-dependent blitting features for faster
+        canvas updates. 
+
+    *lineprops* : dict, default is None
+      Dictionary of :class:`matplotlib.lines.Line2D` properties
+
+    *markerprops* : dict, default is None
+      Dictionary of :class:`matplotlib.markers.MarkerStyle` properties
+
+    *textprops*: dict, default is None
+        Dictionary of :class:`matplotlib.text.Text` properties. To reposition 
+        the textbox you can overide the defaults which position the box in the 
+        top left corner of the axes. 
+
+
+    Usage:
+    ----------
+
+    1. Hold left click drag and release to draw the ruler in the axes.
+      - Hold shift while dragging to lock the ruler to the horizontal axis.
+      - Hold control while drawing to lock the ruler to the vertical axis. 
+
+    2. Right click one of the markers to move the ruler. 
+
+    The keyboard can be used to activate and deactivate the ruler and toggle 
+    visibility of the line and text:
+
+    'm' : Toggles the ruler on and off. 
+
+    'ctl+m' : Toggles the visibility of the ruler and text. 
+
+    Example
+    ----------
+
+    >>> xCoord = np.arange(0, 5, 1)
+    >>> yCoord = [0, 1, -3, 5, -3]
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+
+    >>> markerprops = dict(marker='o', markersize=5, markeredgecolor='red')
+    >>> lineprops = dict(color='red', linewidth=2)
+
+    >>> ax.grid(True)
+    >>> ax.plot(xCoord, yCoord)
+
+    >>> ruler = Ruler(ax=ax,
+                  useblit=True,
+                  markerprops=markerprops,
+                  lineprops=lineprops)
+
+    >>> plt.show()
+
+
+    """
+
+    def __init__(self,
+                 ax,
+                 active=True,
+                 length_unit=None,
+                 angle_unit='degree',
+                 print_text=False,
+                 useblit=False,
+                 lineprops=None,
+                 textprops=None,
+                 markerprops=None):
+        """
+        Add a ruler to *ax*. If ``ruler_active=True``, the ruler will be 
+        activated when the plot is first created. If ``ruler_unit`` is set the 
+        string will be appended to the length text annotations. 
+
+        """
+        AxesWidget.__init__(self, ax)
```

Comment:
How many of the inputs can be passed to the base-class init instead of dealing with them explicitly here?